src/PerfView/GcStats.cs

```diff
@@ -542,43 +543,31 @@ public static void ToXmlAttribs(TextWriter writer, TraceProcess stats, TraceLoad
 
                         if (mt != null)
                         {
-                            writer.Write(" MarkStack =\"{0:n3}", mt.MarkTimes[(int)MarkRootType.MarkStack]);
-                            if (mt.MarkPromoted != null)
+                            void WriteDetailsAboutMarkRootType(MarkRootType type)
                             {
-                                writer.Write("({0})", mt.MarkPromoted[(int)MarkRootType.MarkStack]);
-                            }
-
-                            writer.Write("\" MarkFQ =\"{0:n3}", mt.MarkTimes[(int)MarkRootType.MarkFQ]);
-                            if (mt.MarkPromoted != null)
-                            {
-                                writer.Write("({0})", mt.MarkPromoted[(int)MarkRootType.MarkFQ]);
-                            }
-
-                            writer.Write("\" MarkHandles =\"{0:n3}", mt.MarkTimes[(int)MarkRootType.MarkHandles]);
-                            if (mt.MarkPromoted != null)
-                            {
-                                writer.Write("({0})", mt.MarkPromoted[(int)MarkRootType.MarkHandles]);
-                            }
-
-                            writer.Write("\"");
-                            if (gc.Generation != 2)
-                            {
-                                writer.Write(" MarkOldGen =\"{0:n3}", mt.MarkTimes[(int)MarkRootType.MarkOlder]);
-                                if (mt.MarkPromoted != null)
+                                if (mt.MarkTimes[(int)type] == -1)
                                 {
-                                    writer.Write("({0})", mt.MarkPromoted[(int)MarkRootType.MarkOlder]);
+                                    return;
                                 }
 
-                                writer.Write("\"");
-                            }
-                            if (mt.MarkTimes[(int)MarkRootType.MarkOverflow] != 0.0)
-                            {
-                                writer.Write(" MarkOverflow =\"{0:n3}", mt.MarkTimes[(int)MarkRootType.MarkOverflow]);
+                                writer.Write(" {0} =\"{1:n3}", type, mt.MarkTimes[(int)type]);
                                 if (mt.MarkPromoted != null)
                                 {
-                                    writer.Write("({0})", mt.MarkPromoted[(int)MarkRootType.MarkOverflow]);
+                                    writer.Write("({0})", mt.MarkPromoted[(int)type]);
                                 }
+                                writer.Write("\"");
                             }
+
+                            WriteDetailsAboutMarkRootType(MarkRootType.MarkStack);
```

Comment:
can we just call this in a loop instead of having to duplicate the call many times?